sdcm/cluster_k8s/eks.py

```diff
@@ -334,6 +345,41 @@ def deploy(self):
         self.create_eks_cluster()
         LOGGER.info("Patch kubectl config")
         self.patch_kubectl_config()
+        LOGGER.info("Create storage class")
+        self.create_ebs_storge_class()
+
+    def create_ebs_storge_class(self):
+        if self.params.get('k8s_scylla_disk_class') in SUPPORTED_EBS_STORAGE_CLASSES:
+            tags_specification = "\n".join(
+                [f'  tagSpecification_{i}: "{k}={v}"' for i, (k, v) in enumerate(self.tags.items(), start=1)])
+            self.apply_file(self.STORAGE_CLASS_FILE, environ=dict(EXTRA_TAG_SPEC=tags_specification))
+
+    @property
+    def ebs_csi_driver_info(self) -> dict:
+        addon_info = self.eks_client.describe_addon(
+            clusterName=self.short_cluster_name,
+            addonName='aws-ebs-csi-driver',
+        )
+        LOGGER.debug(pprint.pformat(addon_info))
+        return addon_info['addon']
+
+    @property
+    def ebs_csi_driver_status(self) -> str:
+        return self.ebs_csi_driver_info['status']
+
+    def configure_ebs_csi_driver(self):
+        tags = ",".join([f"{key}={value}" for key, value in self.tags.items()])
+
+        wait_for(lambda: self.ebs_csi_driver_status == 'ACTIVE', step=60, throw_exc=True, timeout=600,
+                 text='Waiting till aws-ebs-csi-driver become operational')
+        LOCALRUNNER.run(
+            f'eksctl utils associate-iam-oidc-provider --region={self.region_name} --cluster={self.short_cluster_name} --approve')
+        LOCALRUNNER.run(f'eksctl create iamserviceaccount --name ebs-csi-controller-sa --namespace kube-system '
+                        f'--cluster {self.short_cluster_name} --attach-policy-arn arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy '
+                        f'--approve --role-name EKS_EBS-{self.short_cluster_name} --region {self.region_name} '
+                        f'--tags {tags} --override-existing-serviceaccounts')
+
+        self.kubectl("rollout restart deployment ebs-csi-controller", namespace="kube-system")
```

Comment:
Why the restart is triggered here? Shouldn't it just start working after eksctl commands executions?

Reply:
I would have thought so as well, but the `eksctl` command just put the configuration in place, and the pods still failing with permission issue.

AWS instructions doesn't state you need the restart, but in the instruction doing the process without eksctl they do mention this step as mandatory

Reply:
Ok